src/app/children/previous-schools/previous-schools.component.ts

```diff
@@ -0,0 +1,88 @@
+import {Component, OnInit, ViewChild} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {DatePipe} from '@angular/common';
+import {ChildSchoolRelation} from '../childSchoolRelation';
+import {ColumnDescription} from '../../ui-helper/entity-subrecord/column-description';
+import {ChildrenService} from '../children.service';
+import {SchoolsService} from '../../schools/schools.service';
+import { School } from 'app/schools/school';
+
+@Component({
+  selector: 'app-previous-schools',
+  template: '<app-entity-subrecord [records]="records" [columns]="columns" [newRecordFactory]="newRecordFactory" [optionalFormValidation]="optionalFormValidation">' + '</app-entity-subrecord>',
+})
+
+
+export class PreviousSchoolsComponent implements OnInit {
+
+  childId: string;
+  records = new Array<ChildSchoolRelation>();
+  schoolList = new Array<School>();
+  columns = new Array<ColumnDescription>();
+
+  constructor(private route: ActivatedRoute,
+              private childrenService: ChildrenService,
+              private schoolsService: SchoolsService,
+              private datePipe: DatePipe) {
+  }
+
+
+  ngOnInit() {
+    this.route.paramMap.subscribe(params => {
+      this.childId = params.get('id').toString();
+      this.loadData(this.childId);
+    });
+  }
+
+
+  loadData(id: string) {
+    let tempRecords = new Array<ChildSchoolRelation>();
+    this.childrenService.getSchoolsWithRelations(id)
+      .then(results => {
+        tempRecords = results;
+        this.records = tempRecords
+          .sort((a, b) => (
+            (b.start ? b.start.valueOf() : 0) - (a.start ? a.start.valueOf() : 0)
+          ));
+      });
+    this.schoolsService.getSchools().subscribe(data => {
+        this.columns = [
+          new ColumnDescription('schoolId', 'Name', 'select',
+            data.map(t => { return { value: t.name, label: t.name} })),
+          new ColumnDescription('start', 'From', 'date', null,
+            (v: Date) => !isNaN(v.getTime()) ? this.datePipe.transform(v, 'yyyy-MM-dd') : undefined), // checking if v is a date and otherweise returning undefined prevents a datePipe error
+          new ColumnDescription('end', 'To', 'date', null,
+            (v: Date) => !isNaN(v.getTime()) ? this.datePipe.transform(v, 'yyyy-MM-dd') : undefined),
+        ];
+    });
+  }
+
+
+  newRecordFactory = () => {
+    const newPreviousSchool = new ChildSchoolRelation(Date.now().toString());
```

Comment:
Instead of using the `Date.now()` please use the `uniqid()` package. To find out how to use it, have a look at the childDetails or schoolDetails component.